scheduler/db/dbmanager.py

```diff
@@ -23,21 +24,42 @@ class DBManager:
     def __init__(self, db_path):
         self.db_path = db_path
 
-    def read(self) -> List[SPlans]:
+
+    def read(self, start_date: Optional[str]=None,
+             end_date: Optional[str]=None,
+             site: Optional[FrozenSet[Site]]=None) -> List[SPlans]:
         with locking(f'{self.db_path}.lock', LOCK_SH):
             with shelve.open(self.db_path) as db:
-                if 'plans' not in db:
-                    db['plans'] = []
-                return db['plans']
-
-    def write(self, plans: List[SPlans]) -> NoReturn:
+                if start_date and end_date and site:
+                    plans_by_site = db[start_date+end_date]
```

Comment:
Not quite sure why you add the `start_date` and the `end_date`? I know if one (or both) are `None`, it's going to be problematic, but are you using the concatenation as an identified for plans?